use-cases/using-SRIOV-and-kata.md

```diff
@@ -0,0 +1,313 @@
+
+2. Find the NICs of interest
+
+   Find PCI details for the NICs in question:
+   ```
+   $ lspci | grep Ethernet
+   00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 04)
+   01:00.0 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   ```
+   In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+   that we will use. You can use `lshw` command to get further details on the 
+   controller and verify it supports SR-IOV.
+
+3. Check how many VFs you can create:
+
+   ```
+   $ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+   63
+   $ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+   63
+   ```
+   The previous commands show how many VFs you can create. The `sriov_totalvfs` 
+   file under sysfs for a PCI device specifies the total number of VFs that you
+   can create.
+
+4. Create the VFs:
+
+   ```
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+   ```
+   
+   Create virtual functions by editing `sriov_numvfs`. In our example, we create
+   commands create virtual functions by editing `sriov_numvfs`. This example 
+   creates one VF per physical device. Note, creating one VF eliminates the 
+   usefulness of SR-IOV, and is done for simplicity in this example. 
+
+ 5. Verify the VFs were added to the host:
+
+   ```
+   # lspci | grep Ethernet | grep Virtual
+   02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   ```
+
+6. Assign a MAC address to each VF:
+   
+   ```
+   $ ip link set <pf> vf <vfidx> mac <fake MAC address>
+   ```
+
+   Depending on the NIC being used, you  might need to explicitly set the MAC 
+   address for the VF device. Setting the MAC address guarantees that the 
+   address is consistent on the host and when passed to the guest. 
+
+## Example: Launch a Kata Containers container using SR-IOV
+
+The following example launches a Kata Containers container using SR-IOV:
+
+1. Build and start SR-IOV plugin:
+
+   To install the SR-IOV plugin, follow the [SR-IOV plugin installation instructions](https://github.com/clearcontainers/sriov) 
+
+2. Create the docker network:
+
+   ```
+   $ sudo docker network create -d sriov --internal --opt pf_iface=enp1s0f0 --opt vlanid=100 --subnet=192.168.0.0/24 vfnet
+
+   E0505 09:35:40.550129    2541 plugin.go:297] Numvfs and Totalvfs are not same on the PF - Initialize numvfs to totalvfs
+   ee2e5a594f9e4d3796eda972f3b46e52342aea04cbae8e5eac9b2dd6ff37b067
+   ```
+   The previous commands create the SR-IOV docker network, a subnet, vlanid, 
+   and a physical interface, all which are required.
+  
+3. Start containers and test their connectivity
```

Comment:
Suggest moving line 297 to follow the example so we can use a colon after step three for consistency (see my note on line 214). Then line 302 would be: The previous example starts a container making use of SR-IOV.